crates/runtime/src/traphandlers.rs

```diff
@@ -207,22 +208,31 @@ pub unsafe fn catch_traps<'a, F>(
     capture_backtrace: bool,
     capture_coredump: bool,
     caller: *mut VMContext,
+    callee: *mut VMOpaqueContext,
     mut closure: F,
 ) -> Result<(), Box<Trap>>
 where
     F: FnMut(*mut VMContext),
 {
     let limits = Instance::from_vmctx(caller, |i| i.runtime_limits());
-
-    let result = CallThreadState::new(signal_handler, capture_backtrace, capture_coredump, *limits)
-        .with(|cx| {
-            traphandlers::wasmtime_setjmp(
-                cx.jmp_buf.as_ptr(),
-                call_closure::<F>,
-                &mut closure as *mut F as *mut u8,
-                caller,
-            )
-        });
+    let callee_stack_cain = VMContext::try_from_opaque(callee)
```

Comment:
```suggestion
    let callee_stack_chain = VMContext::try_from_opaque(callee)
```

Reply:
If I understand this code correctly, then there is a new linear cost to enter Wasm? I suppose the cost may be negligible for now. Nonetheless, I wonder whether we can make this computation lazy? Do we always require immediate access to the `StackChainCell`?

Reply:
I think it is worth documenting for our future selves.

Reply:
Okay, I've extended the existing documentation in `invoke_wasm_and_catch_traps`